base/io.jl

```diff
@@ -1051,6 +1051,114 @@ IteratorSize(::Type{<:EachLine}) = SizeUnknown()
 
 isdone(itr::EachLine, state...) = eof(itr.stream)
 
+# Reverse-order iteration for the EachLine iterator.   (For seekable streams,
+# this reads the stream from the end in 4kB chunks.)
+function iterate(r::Iterators.Reverse{<:EachLine})
+    try
+        seekend(r.itr.stream) # may throw for some stream types
+    catch
```

Comment:
This feels like it should catch some specific exception. Feels weird if this `catch` would "take" if one has e.g. an `UndefVarError` in the `seekend` function.

Reply:
Sure, but what exceptions?   At least `MethodError` and `SystemError`, it seems?

Reply:
I dunno, if it isn't a clear answer, perhaps there is something that is missing (a trait?) or that this method is currently too wide.

Reply:
I've re-implemented it in a more conservative way via `_maybe_seekend`.

(I currently don't see any generic way to check whether a stream is seekable.)